src/ConnectedMode.UnitTests/Migration/XmlFileCleanerTests.cs

```diff
@@ -26,10 +27,16 @@
 
 namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.Migration
 {
+    // NOTE: in the event of a test failure the actual and expected results are
+    // written to files and added to the Test Results i.e. they should be
+    // accessible at the bottom of the test results pane (and automatically
+    // captured by the Azure DevOps CI build).
+
```

Comment:
Writing the fix was easy: writing the tests was a pain because the whitespace and indentation in the "cleaned" file have to be exactly right.

So, I wrote some helper methods to simplify seeing why the files are different.
The files are written to disk and added to the tests results, which means you can access the files from the test result window:

![image](https://github.com/SonarSource/sonarlint-visualstudio/assets/10757559/23e4c249-e3a9-421b-aecf-fa442dafeb87)

Clicking on the file name opens it. The `actual` and `expected` files are next to each other on disc so you can easily diff them.